src/Transformation.tsx

```diff
@@ -151,11 +109,11 @@ function Transformation(): ReactElement {
             setErrMsg("Please apply transformation to a new table first.");
             return;
           }
-          setContextItems(lastContextName, newData);
+          setContextItems(lastContextName, filtered.records);
         } else {
           const [newContext, _newTable] = await createTableWithData(
             inputDataCtxt,
-            newData
+            filtered.records
           );
           setLastContextName(newContext.name);
         }
```

Comment:
Note that we'll probably need to change the way we generate transformation output, as opposed to just setting context items or creating a new table with the new items, because transformations may change the structure of collections.

On the CODAP phone end, we should probably add something like `createDataset` which creates a new data context of the given structure, and then also adds all the given items into it (as opposed to assuming only the items change and the rest can be cloned from the input data context by name). Does that seem reasonable? @stardust66 @P-bibs

Reply:
Yeah, that makes sense. Maybe we'd want to set up some kind of diff so that we know which attributes are new and can add them. We want to clone old attributes sometimes, like if an attribute is computed, we want to make sure the formula is copied. This introduces complications like if the user changes a computed column with map, we need to make sure we don't copy the old formula.